core/api/src/services/auth/role-checker.ts

```diff
@@ -0,0 +1,104 @@
+import { GoogleAuth } from "google-auth-library"
+
+import { env } from "../../config/env"
+
+// Based on https://cloud.google.com/resource-manager/reference/rest/v1/Policy
+interface Policy {
+  bindings?: Binding[]
+  etag?: string
+  version?: number
+}
+
+interface Binding {
+  role?: string
+  members?: string[]
+  condition?: {
+    title?: string
+    description?: string
+    expression?: string
+  }
+}
+
+export enum AdminRole {
+  VIEWER = "roles/adminPanelViewer",
+  SUPPORT = "roles/adminPanelSupport",
+  ADMIN = "roles/adminPanelAdmin",
+}
+
+export enum AdminFeature {
+  VIEW_ACCOUNTS = "VIEW_ACCOUNTS",
+  MODIFY_ACCOUNTS = "MODIFY_ACCOUNTS",
+  DELETE_ACCOUNTS = "DELETE_ACCOUNTS",
+  VIEW_TRANSACTIONS = "VIEW_TRANSACTIONS",
+  SEND_NOTIFICATIONS = "SEND_NOTIFICATIONS",
+  SYSTEM_CONFIG = "SYSTEM_CONFIG",
+}
+
+const ROLE_FEATURES = {
+  [AdminRole.VIEWER]: [AdminFeature.VIEW_ACCOUNTS, AdminFeature.VIEW_TRANSACTIONS],
+  [AdminRole.SUPPORT]: [
+    AdminFeature.VIEW_ACCOUNTS,
+    AdminFeature.MODIFY_ACCOUNTS,
+    AdminFeature.VIEW_TRANSACTIONS,
+    AdminFeature.SEND_NOTIFICATIONS,
+  ],
+  [AdminRole.ADMIN]: Object.values(AdminFeature),
+}
+
+export class RoleChecker {
```

Comment:
we dont use classes, please use the same code style as other services (including arrow function expressions).. the same apply to interfaces and types